### ISSUE TITLE
Bug in EMP_alpha(int precision)?

### ISSUE THREAD

**Issue by astolman** [dialtr/libcount#12]
In your alpha_m function in empirical_data.cc, you parameterize it by the precision, but in the paper, the authors have it parameterized by the actual register count, (2 ^ precision).

**Comment by dialtr** [comment]
Thanks for the comment, I will take a look ASAP.

**Comment by dialtr** [comment]
Hi,

I think I see the error.  Line 59 of empirical_data.cc is this:

return (0.7213 / (1.0 + (1.079 / static_cast<double>(precision))));

It should be:

return (0.7213 / (1.0 + (1.079 / static_cast<double>(1 << precision))));

Per the pseudocode on the top-left hand side of the paper on page 8:

    αm := 0.7213/(1 + 1.079/m) for m ≥ 128

Is that what you had in mind?  Your review is much appreciated. How did you
land on my Github?  Are you using libcount for anything?

dialtr@gmail.com
Tom Dial

**Comment by astolman** [comment]
That is exactly what I had in mind. I found it while googling hyperlog++ implementations, hoping to find the reference implementation from that publication, but yours is the only one in C++ I could find.

**Comment by dialtr** [comment]
Well, thanks again. I posted a PR - I think it ended up as issue #13.  Take a look if you have time. 

If you'd like me to modify the README to cite your contribution, let me know. I didn't want to do so without asking.  If so, I'd probably just put your name and whatever email you'd prefer.  

I learned about HyperLogLog++ at an ElasticSearch conference and thought it was neat. Like you, I poked around for a C++ implementation but couldn't find one. 

I haven't gone all the way with it yet (no sparse register storage) but hope that you found the code readable and useful. 

Cheers,
Tom
